exist-core/src/main/java/org/exist/util/Configuration.java

```diff
@@ -887,558 +929,472 @@ private void configureBackend( final Optional<Path> dbHome, Element con ) throws
-                    final Element trigger = (Element)nlTrigger.item(i);
-                    
-                    // Get @class
-                    final String startupTriggerClass = trigger.getAttribute("class");
-                    
-                    boolean isStartupTrigger = false;
-                    try {
-                        // Verify if class is StartupTrigger
-                        for(final Class iface : Class.forName(startupTriggerClass).getInterfaces()) {
-                            if("org.exist.storage.StartupTrigger".equals(iface.getName())) {
-                                isStartupTrigger = true;
-                                break;
-                            }
-                        }
+        // Iterate over <trigger> elements
+        for (int i = 0; i < nlTrigger.getLength(); i++) {
 
-                        // if it actually is a StartupTrigger
-                        if(isStartupTrigger) {
-                            // Parse additional parameters
-                            final Map<String, List<? extends Object>> params 
-                                    = ParametersExtractor.extract(trigger.getElementsByTagName(ParametersExtractor.PARAMETER_ELEMENT_NAME));
-                            
-                            // Register trigger
-                            startupTriggers.add(new StartupTriggerConfig(startupTriggerClass, params));
-                            
-                            // Done
-                            LOG.info("Registered StartupTrigger: {}", startupTriggerClass);
-                            
-                        } else {
-                            LOG.warn("StartupTrigger: {} does not implement org.exist.storage.StartupTrigger. IGNORING!", startupTriggerClass);
-                        }
-                        
-                    } catch(final ClassNotFoundException cnfe) {
-                        LOG.error("Could not find StartupTrigger class: {}. {}", startupTriggerClass, cnfe.getMessage(), cnfe);
+            // Get <trigger> element
+            final Element trigger = (Element) nlTrigger.item(i);
+
+            // Get @class
+            final String startupTriggerClass = trigger.getAttribute("class");
+
+            boolean isStartupTrigger = false;
+            try {
+                // Verify if class is StartupTrigger
+                for (final Class iface : Class.forName(startupTriggerClass).getInterfaces()) {
+                    if ("org.exist.storage.StartupTrigger".equals(iface.getName())) {
+                        isStartupTrigger = true;
+                        break;
                     }
                 }
+
+                // if it actually is a StartupTrigger
+                if (isStartupTrigger) {
+                    // Parse additional parameters
+                    final Map<String, List<?>> params = ParametersExtractor.extract(trigger.getElementsByTagName(ParametersExtractor.PARAMETER_ELEMENT_NAME));
```

Comment:
This `params` used to be of type `List<? extends Object>>`, changing this from a non-reified to a reified type, will break compiled compatibility with any existing Startup Triggers that are out there in the wild (e.g. EXPath packages etc); this is because it is used as the Interface in `StartupTrigger#execute`.

Please change this back.

Reply:
see above